lib/archethic/bootstrap.ex

```diff
@@ -206,12 +213,91 @@ defmodule Archethic.Bootstrap do
       Logger.info("Synchronization finished")
     end
 
+    persist_gen_addr()
     Sync.publish_end_of_sync()
     SelfRepair.start_scheduler()
 
     :persistent_term.put(:archethic_up, :up)
   end
 
+  # case1: When a node joins a already initialized network
+  # after syncing all transactions persist the gen_addr
+  # case2: node rejoins and requires sync, at end of sync persist gen_addr
+  # case3: first time network initialization
+  def persist_gen_addr() do
+    # first time boot of a node
+    # persist first address
+    Task.Supervisor.async_nolink(Archethic.TaskSupervisor, fn ->
+      # TODO: impl get_last_address_by_tpe
+      # TODO: Get First Address by type
+
+      Task.Supervisor.async_nolink(Archethic.TaskSupervisor, fn ->
+        # init the memtable with the first address of oracle chain according to the summary date
+        summary_interval =
+          Application.get_env(:archethic, Archethic.OracleChain.Scheduler)
+          |> Keyword.fetch(:summary_interval)
+          |> elem(1)
+
+        current_time = DateTime.truncate(DateTime.utc_now(), :second)
+
+        cron_expression = Crontab.CronExpression.Parser.parse!(summary_interval, true)
+
+        naive_date =
+          current_time
+          |> DateTime.truncate(:second)
+          |> DateTime.to_naive()
+
+        next_summary_date =
+          if Crontab.DateChecker.matches_date?(cron_expression, naive_date) do
+            cron_expression
+            |> Crontab.Scheduler.get_next_run_dates(naive_date)
+            |> Enum.at(1)
+            |> DateTime.from_naive!("Etc/UTC")
+          else
+            cron_expression
+            |> Crontab.Scheduler.get_next_run_date!(naive_date)
+            |> DateTime.from_naive!("Etc/UTC")
+          end
+
+        next_summary_date_gen_addr = Crypto.derive_oracle_address(next_summary_date, 0)
+        OracleChain.put_gen_addr(next_summary_date_gen_addr, next_summary_date)
+
+        # current_time = DateTime.utc_now() |> DateTime.truncate(:second)
+        # |> OracleChain.persist_gen_addr(:node_shared_secrets)
+      end)
+
+      Task.Supervisor.async_nolink(Archethic.TaskSupervisor, fn ->
+        TransactionChain.list_addresses_by_type(:mint_rewards)
+        |> Enum.at(0)
+        |> TransactionChain.get_genesis_address()
+        |> Reward.persist_gen_addr()
+      end)
+
+      Task.Supervisor.async_nolink(Archethic.TaskSupervisor, fn ->
+        TransactionChain.list_addresses_by_type(:origin)
+        |> Enum.at(0)
+        |> TransactionChain.get_genesis_address()
+        |> SharedSecrets.persist_gen_addr(:origin)
+      end)
+
+      Task.Supervisor.async_nolink(Archethic.TaskSupervisor, fn ->
```

Comment:
I think we don't need to control `node` transaction as there will be many node transaction chain.